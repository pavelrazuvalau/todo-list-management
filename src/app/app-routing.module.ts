import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { NotFoundPageComponent } from './core/pages/not-found-page/not-found-page.component';

const routes: Routes = [
  { path: '', redirectTo: 'todos', pathMatch: 'full' },
  { path: 'todos', loadChildren: () => import('./todo/todo.module').then(m => m.TodoModule) },
  { path: '**', component: NotFoundPageComponent },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
