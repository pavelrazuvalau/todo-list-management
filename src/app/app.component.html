<header>
  <h1 [innerText]="appTitle"></h1>
</header>

<main>
  <todolist-add-todo (addTodo)="onAddTodo($event)"></todolist-add-todo>
  <todolist-todo-list [todoList]="todoList"></todolist-todo-list>
</main>

<footer></footer>
