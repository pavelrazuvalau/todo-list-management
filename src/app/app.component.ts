import { Component } from '@angular/core';
import { TodoItem } from './models/todo.model';

@Component({
  selector: 'todolist-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  todoList: TodoItem[] = [];
  appTitle = 'Todo list management app';

  constructor() {

  }

  onAddTodo(item: TodoItem) {
    this.todoList = [...this.todoList, item];
  }
}
