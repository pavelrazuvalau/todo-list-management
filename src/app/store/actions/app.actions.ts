import { createAction, props } from '@ngrx/store';

export const setLoadingState = createAction('[App] Set loading', props<{ isLoading: boolean }>());
