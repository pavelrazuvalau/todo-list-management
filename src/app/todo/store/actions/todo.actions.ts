import { createAction, props } from '@ngrx/store';
import { TodoItem } from '../../models/todo.model';

export const loadTodos = createAction('[Todos] Load');
export const loadTodosSuccess = createAction('[Todos] Load Success', props<{ todos: TodoItem[] }>());
export const loadTodosFailed = createAction('[Todos] Load Failed');

export const addTodo = createAction('[Todos] Add todo', props<{ todo: TodoItem }>());
export const addTodoSuccess = createAction('[Todos] Add todo Success', props<{ todo: TodoItem }>());
export const addTodoFailed = createAction('[Todos] Add todo Failed');

export const toggleTodo = createAction('[Todos] Toggle complete', props<{ todo: TodoItem }>());
