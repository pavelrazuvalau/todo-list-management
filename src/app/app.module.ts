import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { AddTodoComponent } from './components/add-todo/add-todo.component';
import { TodoListComponent } from './components/todo-list/todo-list.component';
import { TodoItemComponent } from './components/todo-item/todo-item.component';

@NgModule({
  declarations: [
    AppComponent,
    AddTodoComponent,
    TodoListComponent,
    TodoItemComponent
  ],
  imports: [
    BrowserModule,
    FormsModule,
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
