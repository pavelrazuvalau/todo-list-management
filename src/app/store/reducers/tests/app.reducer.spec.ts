import { initialState, appReducer } from '../app.reducer';
import * as AppActions from '../../actions/app.actions';

describe('appReducer', () => {
  it('should set loading state', () => {
    const action = AppActions.setLoadingState({ isLoading: true });

    const result = appReducer(initialState, AppActions.setLoadingState({ isLoading: true }));

    expect(result).toMatchSnapshot();
  });
});
