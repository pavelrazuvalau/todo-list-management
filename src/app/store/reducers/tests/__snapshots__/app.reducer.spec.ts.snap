// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appReducer should set loading state 1`] = `
Object {
  "isLoading": true,
}
`;
